core/common/modules/connect/apps/base-app.php

```diff
@@ -501,7 +501,8 @@ protected function http_request( $method, $endpoint, $args = [], $options = [] )
 			if ( 401 === $code ) {
 				$this->delete();
 
-				if ( 'xhr' !== $this->auth_mode ) {
+				// Should try to re authorize only in `popup` and default mode.
+				if ( ! in_array( $this->auth_mode, [ 'xhr', 'cli' ], true ) ) {
```

Comment:
Maybe extract to a variable, so you won't need the comment?
IDK, maybe `$retry_modes`?